src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs

```diff
@@ -412,34 +413,131 @@ private protected override SchemaShape.Column[] GetOutputColumnsCore(SchemaShape
             return converged;
         }
 
-        private protected override MulticlassLogisticRegressionModelParameters CreatePredictor(VBuffer<float>[] weights, float[] bias)
+        private protected override void CheckLabel(RoleMappedData examples, out int weightSetCount)
+        {
+            examples.CheckMulticlassLabel(out weightSetCount);
+        }
+
+        private protected override float[] InitializeFeatureNormSquared(int length)
+        {
+            Contracts.Assert(0 < length & length <= Utils.ArrayMaxSize);
+            return new float[length];
+        }
+
+        private protected override float GetInstanceWeight(FloatLabelCursor cursor)
+        {
+            return cursor.Weight;
+        }
+    }
+
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> for training a maximum entropy classification model using the stochastic dual coordinate ascent method.
+    /// The trained model <see cref="MaximumEntropyModelParameters"/> produces probabilities of classes.
+    /// </summary>
+    /// <include file='doc.xml' path='doc/members/member[@name="SDCA_remarks"]/*' />
+    public sealed class SdcaMulticlassClassificationTrainer : SdcaMulticlassClassificationTrainerBase<MaximumEntropyModelParameters>
+    {
+        public class Options : CommonOptions
```

Comment:
>public class Options  [](start = 8, length = 21)

sealed?

Reply:
Sure. Thanks.

---
In reply to: [266991361](https://github.com/dotnet/machinelearning/pull/2976#discussion_r266991361) [](ancestors = 266991361)